File: OCH_Tao/ABC/B/362.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [xA, yA] = input[0].split(" ").map(x => parseInt(x, 10));
const [xB, yB] = input[1].split(" ").map(x => parseInt(x, 10));
const [xC, yC] = input[2].split(" ").map(x => parseInt(x, 10));
const AB = (xA - xB) ** 2 + (yA - yB) ** 2;
const BC = (xB - xC) ** 2 + (yB - yC) ** 2;
const CA = (xC - xA) ** 2 + (yC - yA) ** 2;
if (AB + BC === CA || BC + CA === AB || CA + AB === BC) {
  console.log("Yes");
} else {
  console.log("No");
}

File: OCH_Tao/ABC/B/362.py
A = list(map(int,input().split()))
B = list(map(int,input().split()))
C = list(map(int,input().split()))
if (A[0]==B[0] and B[1]==C[1]) or (A[1]==B[1] and B[0]==C[0]) or (B[0]==C[0] and C[1]==A[1]) or (B[1]==C[1] and C[0]==A[0]) or (C[0]==A[0] and A[1]==B[1]) or (C[1]==A[1] and A[0]==B[0]):
  print("Yes")
else:
  if (A[1]-B[1])/(A[0]-B[0])+(B[0]-C[0])/(B[1]-C[1])==0:
    print("Yes")
  elif (B[1]-C[1])/(B[0]-C[0])+(C[0]-A[0])/(C[1]-A[1])==0:
    print("Yes")
  elif (C[1]-A[1])/(C[0]-A[0])+(A[0]-B[0])/(A[1]-B[1])==0:
    print("Yes")
  else:
    print("No")